server/test/repository/ApplicationStatusesRepositoryTest.java

```diff
@@ -0,0 +1,87 @@
+          .setStatusText("Approved")
+          .setLocalizedStatusText(LocalizedStrings.withDefaultValue("Approved"))
+          .build();
+  ApplicationStatusesRepository repo;
+
+  @Before
+  public void setupStatusRepository() {
+    repo = instanceOf(ApplicationStatusesRepository.class);
+  }
+
+  @Test
+  public void canQueryForActiveApplicationStatuses() {
+    // setup
+    Long uniqueProgramId = new Random().nextLong();
+    ProgramModel program =
+        ProgramBuilder.newActiveProgram("Active status tests" + uniqueProgramId, "description")
+            .build();
+    String programName = program.getProgramDefinition().adminName();
+    StatusDefinitions statusDefinitions = new StatusDefinitions(ImmutableList.of(APPROVED_STATUS));
+    // test
+    ApplicationStatusesModel applicationStatusesModel =
+        new ApplicationStatusesModel(
+            programName, statusDefinitions, StatusDefinitionsLifecycleStage.ACTIVE);
+    applicationStatusesModel.save();
+    applicationStatusesModel.setCreateTimeForTest("2041-01-01T00:00:00Z").save();
+    StatusDefinitions statusDefinitionsResult = repo.lookupActiveStatusDefinitions(programName);
+    // assert
+    assertThat(statusDefinitionsResult.getStatuses().size()).isEqualTo(1);
+    assertThat(statusDefinitionsResult.getStatuses().get(0).statusText()).isEqualTo("Approved");
+  }
+
+  @Test
+  public void lookupActiveStatusDefinitions_throwsException() {
+    assertThatExceptionOfType(RuntimeException.class)
+        .isThrownBy(() -> repo.lookupActiveStatusDefinitions("random"))
+        .withMessage("No active status found for program random");
+  }
+
+  @Test
+  public void lookupListOfObsoleteStatusDefinitions_throwsException() {
+    assertThatExceptionOfType(RuntimeException.class)
+        .isThrownBy(() -> repo.lookupListOfObsoleteStatusDefinitions("random"))
+        .withMessage("No obsolete status found for program random");
+  }
+
+  @Test
+  public void canQueryForOboseleteApplicationStatuses() {
+    Long uniqueProgramId = new Random().nextLong();
+    ProgramModel program =
+        ProgramBuilder.newActiveProgram("test program" + uniqueProgramId, "description").build();
+    String programName = program.getProgramDefinition().adminName();
+    StatusDefinitions statusDefinitions = new StatusDefinitions(ImmutableList.of(APPROVED_STATUS));
+
+    ApplicationStatusesModel applicationStatusesModel =
+        new ApplicationStatusesModel(
+            programName, statusDefinitions, StatusDefinitionsLifecycleStage.OBSOLETE);
+    applicationStatusesModel.save();
+    applicationStatusesModel.setCreateTimeForTest("2041-01-01T00:00:00Z").save();
+    List<StatusDefinitions> statusDefinitionsResults =
+        repo.lookupListOfObsoleteStatusDefinitions(programName);
+
+    assertThat(statusDefinitionsResults).isNotEmpty();
+    assertThat(statusDefinitionsResults.size()).isEqualTo(1);
+    assertThat(statusDefinitionsResults.get(0).getStatuses().get(0).statusText())
```

Comment:
**suggestion, not blocking**: For symmetry/completeness, we could also add a
```
assertThat(statusDefinitionsResults.get(0).getStatuses().size()).isEqualTo(1);
```
to make sure the `StatusDefinitions` has the right number of statuses in it.

Reply:
Done!